src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2400,8 +2472,6 @@ private void StartNestedResourceInfo(ODataJsonLightReaderNestedResourceInfo read
         private void StartNestedPropertyInfo(ODataJsonLightReaderNestedPropertyInfo readerNestedPropertyInfo)
         {
             Debug.Assert(readerNestedPropertyInfo != null, "readerNestedResourceInfo != null");
-            Debug.Assert(this.jsonLightResourceDeserializer.JsonReader.CanStream() || this.CurrentScope is JsonLightResourceSetScope,

```

Comment:
Sacrificed this debug assert statement. Otherwise I'd need to support 2 variants; the sync variant that calls `CanStream` method and the async variant that calls `CanStreamAsync`. That would mean adding an equivalent `StartNestedPropertyInfoAsync` method on account of a debug assert statement!